### PR TITLE
More ground work

### PR DESCRIPTION
**Pull request by cmaion**
- Resiliency improvements
- LR->PD metadata sync
- Don't upload as new images that have already been published in another gallery
- API calls optimizations
- PD->LR gallery structure sync
- Account switching fixes

### CODE REVIEW

**Comment by cmaion** [comment]
At this point, the plugin should be quite robust and hopefully avoid doing dangerous stuff :)

**Comment by willthames** [comment]
Hi @cmaion, just wanted to let you know I'm not ignoring this PR, just need to get around to testing it, sorry! Hopefully will be able to merge soon.

**Comment by willthames** [comment]
We've tested 0.10.1. So we're a bit behind. Would be good to have sign off from at least one other person who is using it as well. In the worst case we can still can sign this off but would be good to have some additional independent testing.

File: Info.lua [hunk 1]
```diff
@@ -2,12 +2,13 @@ return {
     LrSdkVersion = 3.0,
     LrSdkMinimumVersion = 3.0, -- minimum SDK version required by this plug-in
     LrToolkitIdentifier = 'au.id.willthames.photodeck',
-    LrPluginName = 'PhotoDeck Publisher',
+    LrPluginName = 'PhotoDeck',
```
Comment (willthames) [inline Info.lua]: Will this change cause problems for people using existing plugin? Bit reluctant to cause breaking changes for long term users.

Comment (cmaion) [inline Info.lua]: I don't believe this is part of the plugin key in the Lightroom catalog. Don't remember seeing any effect after having changed this either.
But we can revert that commit if you don't like this change ;)
The rational behind this change was that the plugin can also be used for simple export operations (so the name "PhotoDeck Published" was a bit confusing in that case).

**Comment by junction10** [comment]
Tested 0.11.2 @cmaion - Seems great..  pulls in existing gallery structure fine.

Couple of things:

Gallery sync - if I rename a gallery on the photodeck website, how or when will the change get synced in the photodeck plugin?  If I right-click the gallery in Lightroom and choose 'edit gallery', it immediately picks up the revised name, as it stands from the website..  but is it possible for that to be automatically reflected in Lightroom?

Also - if I move the location of a gallery on the website - it gets lost inside Lightroom..  ie: clicking 'Go to Photodeck Gallery' results in page not found.

What would happen if I move a gallery from within Lightroom, from under one parent folder to another for example - would this change be reflected next sync?

**Comment by cmaion** [comment]
Thanks @junction10 for your tests & feedback!

Gallery sync: for now, the sync task is only executed on demand, as it can be quite heavy for large sites. As you have found out by yourself, the Lightroom/PhotoDeck plugin will always show the current gallery name when going to it's settings (and saving the settings should rename it within Lightroom). Or you can manually start a full gallery re-synchronization.

Moving a gallery: yes indeed, the "Go to PhotoDeck Gallery" links don't get properly updated when moving stuff around. A full gallery re-synchronization should fix that.

Moving a gallery from Lightroom: change should be immediate (if I remember correctly).

**Comment by cmaion** [comment]
Feedback from users is good. Merging!

**Comment by willthames** [comment]
That's great @cmaion - thanks!

**Comment by cmaion** [comment]
Thanks to you @willthames ;)

**Pull request merged**